transport/transport.ts

```diff
@@ -835,7 +921,12 @@ export abstract class ServerTransport<
         reason,
       });
       conn.send(this.codec.toBuffer(responseMsg));
-      log?.warn(`${reason}: ${JSON.stringify(parsed)}`, {
+      // note: do _not_ log the payload, it may contain secrets
+      const logData =
+        typeof parsed.payload === 'object' && parsed.payload !== null
```

Comment:
```suggestion
        typeof parsed.payload === 'object'
```

we already check falsey values above :shrug: